editoast/src/schema/rolling_stock_image.rs

```diff
@@ -0,0 +1,211 @@
+}
+
+#[derive(Debug, Identifiable, Queryable, Serialize)]
+#[diesel(belongs_to(RollingStockLivery, foreign_key = livery_id))]
+#[diesel(table_name = osrd_infra_rollingstockimage)]
+pub struct RollingStockSeparatedImage {
+    id: i64,
+    image: Vec<u8>, // binary field
+    livery_id: i64,
+    order: i64,
+}
+
+#[derive(Debug, Insertable)]
+#[diesel(table_name = osrd_infra_rollingstockimage)]
+struct RollingStockImageForm {
+    image: Vec<u8>,
+    order: Option<i64>,
+    livery_id: Option<i64>,
+}
+
+#[derive(Debug, Error, EditoastError)]
+#[editoast_error(base_id = "rollingstockliverycompoundimages")]
+enum CompoundImageError {
+    #[error("Compound image {image_id} of rolling stock livery '{livery_id}', could not be found")]
+    #[editoast_error(status = 404)]
+    NotFound { image_id: i64, livery_id: i64 },
+    #[error("Image '{image_id}' of livery {livery_id} cannot be deleted. It is probably used.")]
+    CanNotDelete { image_id: i64, livery_id: i64 },
+}
+
+impl RollingStockCompoundImage {
+    /// Return the owned image
+    pub fn inner_data(self) -> Vec<u8> {
+        self.image
+    }
+
+    /// Return the compound image
+    pub async fn retrieve(
+        db_pool: Data<DbPool>,
+        livery_id: i64,
+        image_id: i64,
+    ) -> Result<RollingStockCompoundImage> {
+        use crate::tables::osrd_infra_rollingstockimage::dsl;
+        block(move || {
+            let mut conn = db_pool.get().expect("Failed to get DB connection");
+            match dsl::osrd_infra_rollingstockimage
+                .find(image_id)
+                .select(RollingStockCompoundImage::as_select())
+                .first(&mut conn)
+            {
+                Ok(livery) => Ok(livery),
+                Err(DieselError::NotFound) => Err(CompoundImageError::NotFound {
+                    image_id,
+                    livery_id,
+                }
+                .into()),
+                Err(e) => Err(e.into()),
+            }
+        })
+        .await
+        .unwrap()
+    }
+
+    /// Create a compound image
+    pub async fn create(db_pool: Data<DbPool>, image: Vec<u8>) -> Result<i64> {
+        let form = RollingStockImageForm {
+            image,
+            order: None,
+            livery_id: None,
+        };
+        block(move || {
+            use crate::tables::osrd_infra_rollingstockimage::dsl;
+            let mut conn = db_pool.get().expect("Failed to get DB connection");
+            match insert_into(dsl::osrd_infra_rollingstockimage)
+                .values(&form)
+                .returning(dsl::id)
+                .get_result(&mut conn)
+            {
+                Ok(image_id) => Ok(image_id),
+                Err(e) => Err(e.into()),
+            }
+        })
+        .await
+        .unwrap()
+    }
+
+    /// Delete a compound image and set the compound_image_id to null in its livery
+    pub async fn delete(db_pool: Data<DbPool>, livery_id: i64, image_id: i64) -> Result<()> {
```

Comment:
You use `livery_id` only for the error message
Furthermore you say "set the compound_image_id to null" is it done automatically ?

Reply:
Oh okay it's done automatically thanks to Django, I find the function's docstring a bit confusing then